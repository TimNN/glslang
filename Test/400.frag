#version 330 core

in vec2 c2D;
flat in int i;
out vec4 outp;
uniform sampler2D arrayedSampler[5];

void main()
{
    vec4 v;
    v = texture(arrayedSampler[i], c2D);
    outp.x = gl_ClipDistance[1];
}
