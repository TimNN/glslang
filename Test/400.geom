#version 400 core

void main()
{
    EmitStreamVertex(1);
    EndStreamPrimitive(0);
    EmitVertex();
    EndPrimitive();

    int id = gl_InvocationID;
}
