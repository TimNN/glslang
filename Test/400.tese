#version 400 core

layout(vertices = 4) out; // ERROR
layout(quads, cw) in;
layout(triangles) in;     // ERROR
layout(isolines) in;      // ERROR

layout(cw) in;
layout(ccw) in;           // ERROR

layout(fractional_odd_spacing) in;    
layout(equal_spacing) in;              // ERROR
layout(fractional_even_spacing) in;    // ERROR

layout(point_mode) in;

void main()
{
    barrier(); // ERROR

    int a = gl_MaxTessEvaluationInputComponents +
            gl_MaxTessEvaluationOutputComponents +
            gl_MaxTessEvaluationTextureImageUnits +
            gl_MaxTessEvaluationUniformComponents +
            gl_MaxTessPatchComponents +
            gl_MaxPatchVertices +
            gl_MaxTessGenLevel;

    vec4 p = gl_in[1].gl_Position;
    float ps = gl_in[1].gl_PointSize;
    float cd = gl_in[1].gl_ClipDistance[2];

    int pvi = gl_PatchVerticesIn;
    int pid = gl_PrimitiveID;
    vec3 tc = gl_TessCoord;
    float tlo = gl_TessLevelOuter[3];
    float tli = gl_TessLevelInner[1];

    gl_Position = p;
    gl_PointSize = ps;
    gl_ClipDistance[2] = cd;
}
