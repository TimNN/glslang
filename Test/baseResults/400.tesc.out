400.tesc
Warning, version 400 is not yet complete; most version-specific features are present, but some are missing.
ERROR: 0:6: 'quads' : unrecognized layout identifier, or qualifier requires assignment (e.g., binding = 4) 
ERROR: 0:7: 'ccw' : unrecognized layout identifier, or qualifier requires assignment (e.g., binding = 4) 
ERROR: 0:8: 'fractional_even_spacing' : unrecognized layout identifier, or qualifier requires assignment (e.g., binding = 4) 
ERROR: 0:10: 'patch' : can only use on output in tessellation-control shader 
ERROR: 0:39: 'vertices' : can only apply to 'out' 
ERROR: 0:40: 'vertices' : cannot change previously set layout value 
ERROR: 0:44: '[' :  array index out of range '4'
ERROR: 0:47: 'in' : type must be an array: ina
ERROR: 0:49: '[]' : tessellation input array size must be gl_MaxPatchVertices or implicitly sized 
ERROR: 0:56: 'location' : overlapping use of location 4
ERROR: 0:60: 'location' : overlapping use of location 4
ERROR: 11 compilation errors.  No code generated.


Shader version: 400
Requested GL_ARB_separate_shader_objects
vertices = 4
ERROR: node is still EOpNull!
0:13  Function Definition: main( (void)
0:13    Function Parameters: 
0:15    Sequence
0:15      Barrier (void)
0:17      Sequence
0:17        move second child to first child (int)
0:17          'a' (int)
0:17          Constant:
0:17            5392 (const int)
0:23      Sequence
0:23        move second child to first child (4-component vector of float)
0:23          'p' (4-component vector of float)
0:23          gl_Position: direct index for structure (in 4-component vector of float)
0:23            direct index (block{in 4-component vector of float gl_Position, in float gl_PointSize, in implicitly-sized array of float gl_ClipDistance})
0:23              'gl_in' (in 32-element array of block{in 4-component vector of float gl_Position, in float gl_PointSize, in implicitly-sized array of float gl_ClipDistance})
0:23              Constant:
0:23                1 (const int)
0:23            Constant:
0:23              0 (const int)
0:24      Sequence
0:24        move second child to first child (float)
0:24          'ps' (float)
0:24          gl_PointSize: direct index for structure (in float)
0:24            direct index (block{in 4-component vector of float gl_Position, in float gl_PointSize, in implicitly-sized array of float gl_ClipDistance})
0:24              'gl_in' (in 32-element array of block{in 4-component vector of float gl_Position, in float gl_PointSize, in implicitly-sized array of float gl_ClipDistance})
0:24              Constant:
0:24                1 (const int)
0:24            Constant:
0:24              1 (const int)
0:25      Sequence
0:25        move second child to first child (float)
0:25          'cd' (float)
0:25          direct index (float)
0:25            gl_ClipDistance: direct index for structure (in implicitly-sized array of float)
0:25              direct index (block{in 4-component vector of float gl_Position, in float gl_PointSize, in implicitly-sized array of float gl_ClipDistance})
0:25                'gl_in' (in 32-element array of block{in 4-component vector of float gl_Position, in float gl_PointSize, in implicitly-sized array of float gl_ClipDistance})
0:25                Constant:
0:25                  1 (const int)
0:25              Constant:
0:25                2 (const int)
0:25            Constant:
0:25              2 (const int)
0:27      Sequence
0:27        move second child to first child (int)
0:27          'pvi' (int)
0:27          'gl_PatchVerticesIn' (in int)
0:28      Sequence
0:28        move second child to first child (int)
0:28          'pid' (int)
0:28          'gl_PrimitiveID' (in int)
0:29      Sequence
0:29        move second child to first child (int)
0:29          'iid' (int)
0:29          'gl_InvocationID' (in int)
0:31      move second child to first child (4-component vector of float)
0:31        gl_Position: direct index for structure (out 4-component vector of float)
0:31          direct index (block{out 4-component vector of float gl_Position, out float gl_PointSize, out implicitly-sized array of float gl_ClipDistance})
0:31            'gl_out' (out 4-element array of block{out 4-component vector of float gl_Position, out float gl_PointSize, out implicitly-sized array of float gl_ClipDistance})
0:31            Constant:
0:31              1 (const int)
0:31          Constant:
0:31            0 (const int)
0:31        'p' (4-component vector of float)
0:32      move second child to first child (float)
0:32        gl_PointSize: direct index for structure (out float)
0:32          direct index (block{out 4-component vector of float gl_Position, out float gl_PointSize, out implicitly-sized array of float gl_ClipDistance})
0:32            'gl_out' (out 4-element array of block{out 4-component vector of float gl_Position, out float gl_PointSize, out implicitly-sized array of float gl_ClipDistance})
0:32            Constant:
0:32              1 (const int)
0:32          Constant:
0:32            1 (const int)
0:32        'ps' (float)
0:33      move second child to first child (float)
0:33        direct index (float)
0:33          gl_ClipDistance: direct index for structure (out implicitly-sized array of float)
0:33            direct index (block{out 4-component vector of float gl_Position, out float gl_PointSize, out implicitly-sized array of float gl_ClipDistance})
0:33              'gl_out' (out 4-element array of block{out 4-component vector of float gl_Position, out float gl_PointSize, out implicitly-sized array of float gl_ClipDistance})
0:33              Constant:
0:33                1 (const int)
0:33            Constant:
0:33              2 (const int)
0:33          Constant:
0:33            1 (const int)
0:33        'cd' (float)
0:35      move second child to first child (float)
0:35        direct index (patch float)
0:35          'gl_TessLevelOuter' (patch out 4-element array of float)
0:35          Constant:
0:35            3 (const int)
0:35        Constant:
0:35          3.200000
0:36      move second child to first child (float)
0:36        direct index (patch float)
0:36          'gl_TessLevelInner' (patch out 2-element array of float)
0:36          Constant:
0:36            1 (const int)
0:36        Constant:
0:36          1.300000
0:42  Function Definition: foo( (void)
0:42    Function Parameters: 
0:44    Sequence
0:44      gl_PointSize: direct index for structure (out float)
0:44        direct index (block{out 4-component vector of float gl_Position, out float gl_PointSize, out implicitly-sized array of float gl_ClipDistance})
0:44          'gl_out' (out 4-element array of block{out 4-component vector of float gl_Position, out float gl_PointSize, out implicitly-sized array of float gl_ClipDistance})
0:44          Constant:
0:44            4 (const int)
0:44        Constant:
0:44          1 (const int)
0:64  Function Definition: foop( (void)
0:64    Function Parameters: 
0:?     Sequence
0:68      multiply second child into first child (3-component vector of float)
0:68        'pv3' (3-component vector of float)
0:68        'pv3' (3-component vector of float)
0:69      move second child to first child (3-component vector of float)
0:69        'pv3' (3-component vector of float)
0:69        Function Call: fma(vf3;vf3;vf3; (3-component vector of float)
0:69          'pv3' (3-component vector of float)
0:69          'pv3' (3-component vector of float)
0:69          'pv3' (3-component vector of float)
0:70      move second child to first child (double)
0:70        'd' (double)
0:70        Function Call: fma(d1;d1;d1; (double)
0:70          'd' (double)
0:70          'd' (double)
0:70          'd' (double)
0:?   Linker Objects
0:?     'gl_out' (out 4-element array of block{out 4-component vector of float gl_Position, out float gl_PointSize, out implicitly-sized array of float gl_ClipDistance})
0:?     'outa' (4-element array of int)
0:?     'patchIn' (patch in 4-component vector of float)
0:?     'patchOut' (patch out 4-component vector of float)
0:?     'ina' (in 2-component vector of float)
0:?     'inb' (in 32-element array of 2-component vector of float)
0:?     'inc' (in 32-element array of 2-component vector of float)
0:?     'ind' (in 32-element array of 2-component vector of float)
0:?     'ivla' (layout(location=3 ) in 32-element array of 4-component vector of float)
0:?     'ivlb' (layout(location=4 ) in 32-element array of 4-component vector of float)
0:?     'ivlc' (layout(location=4 ) in 32-element array of 4-component vector of float)
0:?     'ovla' (layout(location=3 ) out 4-element array of 4-component vector of float)
0:?     'ovlb' (layout(location=4 ) out 4-element array of 4-component vector of float)
0:?     'ovlc' (layout(location=4 ) out 4-element array of 4-component vector of float)
0:?     'pv3' (3-component vector of float)


Linked tessellation control stage:


Shader version: 400
Requested GL_ARB_separate_shader_objects
vertices = 4
ERROR: node is still EOpNull!
0:13  Function Definition: main( (void)
0:13    Function Parameters: 
0:15    Sequence
0:15      Barrier (void)
0:17      Sequence
0:17        move second child to first child (int)
0:17          'a' (int)
0:17          Constant:
0:17            5392 (const int)
0:23      Sequence
0:23        move second child to first child (4-component vector of float)
0:23          'p' (4-component vector of float)
0:23          gl_Position: direct index for structure (in 4-component vector of float)
0:23            direct index (block{in 4-component vector of float gl_Position, in float gl_PointSize, in 1-element array of float gl_ClipDistance})
0:23              'gl_in' (in 32-element array of block{in 4-component vector of float gl_Position, in float gl_PointSize, in 1-element array of float gl_ClipDistance})
0:23              Constant:
0:23                1 (const int)
0:23            Constant:
0:23              0 (const int)
0:24      Sequence
0:24        move second child to first child (float)
0:24          'ps' (float)
0:24          gl_PointSize: direct index for structure (in float)
0:24            direct index (block{in 4-component vector of float gl_Position, in float gl_PointSize, in 1-element array of float gl_ClipDistance})
0:24              'gl_in' (in 32-element array of block{in 4-component vector of float gl_Position, in float gl_PointSize, in 1-element array of float gl_ClipDistance})
0:24              Constant:
0:24                1 (const int)
0:24            Constant:
0:24              1 (const int)
0:25      Sequence
0:25        move second child to first child (float)
0:25          'cd' (float)
0:25          direct index (float)
0:25            gl_ClipDistance: direct index for structure (in 1-element array of float)
0:25              direct index (block{in 4-component vector of float gl_Position, in float gl_PointSize, in 1-element array of float gl_ClipDistance})
0:25                'gl_in' (in 32-element array of block{in 4-component vector of float gl_Position, in float gl_PointSize, in 1-element array of float gl_ClipDistance})
0:25                Constant:
0:25                  1 (const int)
0:25              Constant:
0:25                2 (const int)
0:25            Constant:
0:25              2 (const int)
0:27      Sequence
0:27        move second child to first child (int)
0:27          'pvi' (int)
0:27          'gl_PatchVerticesIn' (in int)
0:28      Sequence
0:28        move second child to first child (int)
0:28          'pid' (int)
0:28          'gl_PrimitiveID' (in int)
0:29      Sequence
0:29        move second child to first child (int)
0:29          'iid' (int)
0:29          'gl_InvocationID' (in int)
0:31      move second child to first child (4-component vector of float)
0:31        gl_Position: direct index for structure (out 4-component vector of float)
0:31          direct index (block{out 4-component vector of float gl_Position, out float gl_PointSize, out 1-element array of float gl_ClipDistance})
0:31            'gl_out' (out 4-element array of block{out 4-component vector of float gl_Position, out float gl_PointSize, out 1-element array of float gl_ClipDistance})
0:31            Constant:
0:31              1 (const int)
0:31          Constant:
0:31            0 (const int)
0:31        'p' (4-component vector of float)
0:32      move second child to first child (float)
0:32        gl_PointSize: direct index for structure (out float)
0:32          direct index (block{out 4-component vector of float gl_Position, out float gl_PointSize, out 1-element array of float gl_ClipDistance})
0:32            'gl_out' (out 4-element array of block{out 4-component vector of float gl_Position, out float gl_PointSize, out 1-element array of float gl_ClipDistance})
0:32            Constant:
0:32              1 (const int)
0:32          Constant:
0:32            1 (const int)
0:32        'ps' (float)
0:33      move second child to first child (float)
0:33        direct index (float)
0:33          gl_ClipDistance: direct index for structure (out 1-element array of float)
0:33            direct index (block{out 4-component vector of float gl_Position, out float gl_PointSize, out 1-element array of float gl_ClipDistance})
0:33              'gl_out' (out 4-element array of block{out 4-component vector of float gl_Position, out float gl_PointSize, out 1-element array of float gl_ClipDistance})
0:33              Constant:
0:33                1 (const int)
0:33            Constant:
0:33              2 (const int)
0:33          Constant:
0:33            1 (const int)
0:33        'cd' (float)
0:35      move second child to first child (float)
0:35        direct index (patch float)
0:35          'gl_TessLevelOuter' (patch out 4-element array of float)
0:35          Constant:
0:35            3 (const int)
0:35        Constant:
0:35          3.200000
0:36      move second child to first child (float)
0:36        direct index (patch float)
0:36          'gl_TessLevelInner' (patch out 2-element array of float)
0:36          Constant:
0:36            1 (const int)
0:36        Constant:
0:36          1.300000
0:42  Function Definition: foo( (void)
0:42    Function Parameters: 
0:44    Sequence
0:44      gl_PointSize: direct index for structure (out float)
0:44        direct index (block{out 4-component vector of float gl_Position, out float gl_PointSize, out 1-element array of float gl_ClipDistance})
0:44          'gl_out' (out 4-element array of block{out 4-component vector of float gl_Position, out float gl_PointSize, out 1-element array of float gl_ClipDistance})
0:44          Constant:
0:44            4 (const int)
0:44        Constant:
0:44          1 (const int)
0:64  Function Definition: foop( (void)
0:64    Function Parameters: 
0:?     Sequence
0:68      multiply second child into first child (3-component vector of float)
0:68        'pv3' (3-component vector of float)
0:68        'pv3' (3-component vector of float)
0:69      move second child to first child (3-component vector of float)
0:69        'pv3' (3-component vector of float)
0:69        Function Call: fma(vf3;vf3;vf3; (3-component vector of float)
0:69          'pv3' (3-component vector of float)
0:69          'pv3' (3-component vector of float)
0:69          'pv3' (3-component vector of float)
0:70      move second child to first child (double)
0:70        'd' (double)
0:70        Function Call: fma(d1;d1;d1; (double)
0:70          'd' (double)
0:70          'd' (double)
0:70          'd' (double)
0:?   Linker Objects
0:?     'gl_out' (out 4-element array of block{out 4-component vector of float gl_Position, out float gl_PointSize, out 1-element array of float gl_ClipDistance})
0:?     'outa' (4-element array of int)
0:?     'patchIn' (patch in 4-component vector of float)
0:?     'patchOut' (patch out 4-component vector of float)
0:?     'ina' (in 2-component vector of float)
0:?     'inb' (in 32-element array of 2-component vector of float)
0:?     'inc' (in 32-element array of 2-component vector of float)
0:?     'ind' (in 32-element array of 2-component vector of float)
0:?     'ivla' (layout(location=3 ) in 32-element array of 4-component vector of float)
0:?     'ivlb' (layout(location=4 ) in 32-element array of 4-component vector of float)
0:?     'ivlc' (layout(location=4 ) in 32-element array of 4-component vector of float)
0:?     'ovla' (layout(location=3 ) out 4-element array of 4-component vector of float)
0:?     'ovlb' (layout(location=4 ) out 4-element array of 4-component vector of float)
0:?     'ovlc' (layout(location=4 ) out 4-element array of 4-component vector of float)
0:?     'pv3' (3-component vector of float)

